discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -84,45 +101,43 @@ def get_host_assigned_roles(self):
     def get_operators(self):
         return self.api_client.get_cluster_operators(self.id)
 
-    def generate_image(self, ssh_key=env_variables['ssh_public_key']):
-        self.api_client.generate_image(cluster_id=self.id, ssh_key=ssh_key)
+    def generate_image(self):
+        self.api_client.generate_image(cluster_id=self.id, ssh_key=self._config.ssh_public_key)
 
     @JunitTestCase()
     def generate_and_download_image(
```

Comment:
removal of ssh_key breaks `set_iptables_rules_for_nodes`

ssh_key should be removed from there as well